pkg/controllers/resources/csidrivers/translate.go

```diff
@@ -0,0 +1,40 @@
+package csidrivers
+
+import (
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+)
+
+func (s *csidriverSyncer) translateBackwards(pCSIDriver *storagev1.CSIDriver) *storagev1.CSIDriver {
+	return s.TranslateMetadata(pCSIDriver).(*storagev1.CSIDriver)
+}
+
+var _ = storagev1.CSIDriver{}
+var _ = storagev1.CSIStorageCapacity{}
+
+func (s *csidriverSyncer) translateUpdateBackwards(pObj, vObj *storagev1.CSIDriver) *storagev1.CSIDriver {
+	var updated *storagev1.CSIDriver
+
+	changed, updatedAnnotations, updatedLabels := s.TranslateMetadataUpdate(vObj, pObj)
+	if changed {
+		updated = newIfNil(updated, vObj)
+		updated.Labels = updatedLabels
+		updated.Annotations = updatedAnnotations
+	}
+
+	// for each csidriver in phyical, translate and copy
+
+	if !equality.Semantic.DeepEqual(vObj.Spec, pObj.Spec) {
+		updated = newIfNil(updated, vObj)
+		pObj.Spec.DeepCopyInto(&updated.Spec)
```

Comment:
I think this is not necessary because `obj.DeepCopy()` used in `newIfNil()` will copy Spec too.

Reply:
Oh nice, will change!

Reply:
It won't copy if the obj to be updated is not nil

Reply:
oh, good point!!